### PR TITLE
update PR tests to run same as deploy time 

### PR DESCRIPTION
**Pull request by stvnjacobs**
## Description

This will flag off deploy steps from the workflow to only run on the
master branch. This way the same tests will run on each PR that will be
run before release.

## Type of Change

- Bug fix ('fix', 'repair', 'bug')

### CODE REVIEW

**Comment by stvnjacobs** [comment]
Another option: We could also split this into two workflows if we don't want to see the skipped tests. In that configuration, the workflow triggered by pull requests would run tests, while the workflow triggered my commits to master would run the builds and deployments.

For future consideration: I also think it may be soon time to consider releasing based off tags instead of every merge to master. In this scenario, we could get rid of the development branch, test PRs as they go into master, and deploy when a version is tagged. It cleans up this pipeline a bit, and also is easier to maintain.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -66,9 +63,9 @@ jobs:
       - run: yarn --frozen-lockfile
       - run: yarn workspace linode-manager run test
 
-  # https://stackoverflow.com/questions/58033366/how-to-get-current-branch-within-github-actions#comment102508135_58034787
   build-manager:
     runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
```
Comment (johnwcallahan) [inline .github/workflows/ci.yml:68]: Will this expression be `true` if a pull request to master is opened?

**Comment by johnwcallahan** [comment]
> Another option: We could also split this into two workflows if we don't want to see the skipped tests. In that configuration, the workflow triggered by pull requests would run tests, while the workflow triggered my commits to master would run the builds and deployments.

I like the idea of always running the tests in the same workflow as the deployment. That way the tests are run if someone pushes directly to master without a PR (which is how we currently do our release).

Personally the skipped checks don't bother me. What'd be really great is if we could have the SDK build + tests as its own workflow that is also run _as part_ of the deployment workflow, but it doesn't seem like that's possible. (In effect that's what's happening here, but it'd be cool if we could put it in its own workflow file).

> For future consideration: I also think it may be soon time to consider releasing based off tags instead of every merge to master. In this scenario, we could get rid of the development branch, test PRs as they go into master, and deploy when a version is tagged. It cleans up this pipeline a bit, and also is easier to maintain.

Yes.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,9 +1,6 @@
-name: release
+name: ci
 
-on:
-  push:
-    branches:
-      - master
+on: [push, pull_request]
```
Comment (Jskobos) [inline .github/workflows/ci.yml:3]: What about something like:

```
on:
  push:
    branches:
      - master
  pull_request:
```

Would do the same thing but clean up the output a bit

**Review by Jskobos — COMMENTED** [review]
Thanks for getting this up so quickly. looks good, and I'm totally on board with removing `develop`

File: .github/workflows/ci.yml (hunk 2 above)
Comment (stvnjacobs) [reply .github/workflows/ci.yml:3]: I believe that would do the same thing. We would still need to check if we are on master in each publish step. Which output are you referencing that would be cleaned up?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (stvnjacobs) [reply .github/workflows/ci.yml:68]: No, the ref would be the current branch, which in this case would be the branch name of the users branch.

https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#github-context

Please do double check me on that. I searched around a bit to find the best way to only run on changes to master, and this was pretty common.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Jskobos) [reply .github/workflows/ci.yml:3]: The steps are all only run once, but technically there's two events for a PR (push to branch and pull request from the branch), so the "pending checks" list has entries for e.g. `push: linode client test` and `pr: linode client test`

Comment (Jskobos) [reply .github/workflows/ci.yml]: Made this change:

Screen Shot 2020-12-01 at 12 25 02 PM

Screen Shot 2020-12-01 at 12 48 40 PM

Comment (stvnjacobs) [reply .github/workflows/ci.yml]: Thank you! Looks good.

**Comment by stvnjacobs** [comment]
Last thing, do we want to build manager on pull requests to make sure it builds? Are there any instances where tests would pass but it would not build successfully?

**Comment by Jskobos** [comment]
> Last thing, do we want to build manager on pull requests to make sure it builds? Are there any instances where tests would pass but it would not build successfully?

Don't we want to deploy the build artifact as we do currently through Jenkins? In general, building isn't critical as long as typechecking and the linter pass. Which...we could include in test-manager but currently don't. If building every PR is going to be hard on the infrastructure or our account limits we can skip it, but it's helpful in general.

**Review by Jskobos — APPROVED** [review]

**Review by ndukatz — APPROVED** [review]

**Review by tiffwong — APPROVED** [review]

**Pull request merged**